imap-proto/src/types/acls.rs

```diff
@@ -0,0 +1,179 @@
+    /// mailbox)
+    Lookup,
+    /// r - read (SELECT the mailbox, perform STATUS)
+    Read,
+    /// s - keep seen/unseen information across sessions (set or clear
+    /// \SEEN flag via STORE, also set \SEEN during APPEND/COPY/
+    /// FETCH BODY[...])
+    Seen,
+    /// w - write (set or clear flags other than \SEEN and \DELETED via
+    /// STORE, also set them during APPEND/COPY)
+    Write,
+    /// i - insert (perform APPEND, COPY into mailbox)
+    Insert,
+    /// p - post (send mail to submission address for mailbox,
+    /// not enforced by IMAP4 itself)
+    Post,
+    /// k - create mailboxes (CREATE new sub-mailboxes in any
+    /// implementation-defined hierarchy, parent mailbox for the new
+    /// mailbox name in RENAME)
+    CreateMailbox,
+    /// x - delete mailbox (DELETE mailbox, old mailbox name in RENAME)
+    DeleteMailbox,
+    /// t - delete messages (set or clear \DELETED flag via STORE, set
+    /// \DELETED flag during APPEND/COPY)
+    DeleteMessage,
+    /// e - perform EXPUNGE and expunge as a part of CLOSE
+    Expunge,
+    /// a - administer (perform SETACL/DELETEACL/GETACL/LISTRIGHTS)
+    Administer,
+    /// n - ability to write .shared annotations values
+    /// From RFC 5257
+    Annotation,
+    /// c - old (deprecated) create. Do not use. Read RFC 4314 for more information.
+    OldCreate,
+    /// d - old (deprecated) delete. Do not use. Read RFC 4314 for more information.
+    OldDelete,
+    /// A custom right
+    Custom(char),
+}
+
+impl From<char> for AclRight {
+    fn from(c: char) -> Self {
+        match c {
+            'l' => AclRight::Lookup,
+            'r' => AclRight::Read,
+            's' => AclRight::Seen,
+            'w' => AclRight::Write,
+            'i' => AclRight::Insert,
+            'p' => AclRight::Post,
+            'k' => AclRight::CreateMailbox,
+            'x' => AclRight::DeleteMailbox,
+            't' => AclRight::DeleteMessage,
+            'e' => AclRight::Expunge,
+            'a' => AclRight::Administer,
+            'n' => AclRight::Annotation,
+            'c' => AclRight::OldCreate,
+            'd' => AclRight::OldDelete,
+            _ => AclRight::Custom(c),
+        }
+    }
+}
+
+impl From<AclRight> for char {
+    fn from(right: AclRight) -> Self {
+        match right {
+            AclRight::Lookup => 'l',
+            AclRight::Read => 'r',
+            AclRight::Seen => 's',
+            AclRight::Write => 'w',
+            AclRight::Insert => 'i',
+            AclRight::Post => 'p',
+            AclRight::CreateMailbox => 'k',
+            AclRight::DeleteMailbox => 'x',
+            AclRight::DeleteMessage => 't',
+            AclRight::Expunge => 'e',
+            AclRight::Administer => 'a',
+            AclRight::Annotation => 'n',
+            AclRight::OldCreate => 'c',
+            AclRight::OldDelete => 'd',
+            AclRight::Custom(c) => c,
+        }
+    }
+}
+
+pub(crate) type AclRightList = Vec<AclRight>;
```

Comment:
I'd prefer to avoid the alias; IMO these usually make it harder to understand the code.

Reply:
yeah.. sorry that was supposed to be removed..   I have it removed now.